packages/serve/src/index.ts

```diff
@@ -10,22 +10,33 @@ import argsToCamelCase from './args-to-camel-case';
  * @param {String[]} args - args processed from the CLI
  * @returns {Function} invokes the devServer API
  */
-export default function serve(...args): void {
+export default function serve(): void {
     const cli = new WebpackCLI();
     const core = cli.getCoreFlags();
-    // partial parsing usage: https://github.com/75lb/command-line-args/wiki/Partial-parsing
 
-    // since the webpack flags have the 'entry' option set as it's default option,
-    // we need to parse the dev server args first. Otherwise, the webpack parsing could snatch
-    // one of the dev server's options and set it to this 'entry' option.
-    // see: https://github.com/75lb/command-line-args/blob/master/doc/option-definition.md#optiondefaultoption--boolean
-    const devServerArgs = cli.commandLineArgs(devServer, { argv: args, partial: true });
-    const webpackArgs = cli.commandLineArgs(core, { argv: devServerArgs._unknown || [], stopAtFirstUnknown: false });
-    const finalArgs = argsToCamelCase(devServerArgs._all || {});
+    const filteredArgs = process.argv.filter((arg) => arg != 'serve');
```

Comment:
Maybe we can implement API method on webpack-cli side, like getFlagsForCommand('serve') and it is remove unnecessary argv?

Reply:
Will be great to avoid working with argv on command side and process on command side. What do you think?

Reply:
actually this is not needed, it seems the filtered args are already passed into the serve command and I just wasn't aware of it